### PR TITLE
Chisel and Statue Improvements

### LINKED ISSUE #65701: Chisel and block not working most of the time.
Reporting client version: 514.1583

## Round ID:
[180555](https://scrubby.melonmesa.com/round/180555)

## Testmerges:
- [Pingbus changes and improvements](https://www.github.com/tgstation/tgstation/pull/65625)
- [Turbine Rework](https://www.github.com/tgstation/tgstation/pull/65586)

## Reproduction:
Most of the time the chisel and block don't work, the theme menu never comes up and nothing happens after getting the message " you prepare to work on the gold block" once I managed to get it working but most of the time it doesn't I asked other players to do it and some can do it and some others can't. this is frustrating.

Steps to reproduce:
1) Craft a block with any material.
2) Use a chisel on the block
3) nothing happens.

### LINKED ISSUE #72607: Carving Blocks can't be carved
Reporting client version: 514.1582

## Round ID:
[197950](https://scrubby.melonmesa.com/round/197950)

## Reproduction:
Attempting to make a statue from a carving block lets you click it the first time to "prepare to work on the MATERIAL block.", but you can't click a different item afterwards to select a subject, meaning no making a statue. Attempted to make statues with multiple materials (Iron, Glass, Titanium, Sheetzas), as well as to have multiple different people try, but it did not work in any combination.

### PR DESCRIPTION
**Pull request by timothymtorres**
## About The Pull Request
Resolves #72607
Resolves #65701

Chisels and statues had quite a few bugs and they lacked any kind of feedback system to inform the user on it's proper usage.  My improvements do the following:

- Chisels now give balloon alerts when selecting a target, cancelling, or sculpting
- Chisels used on carving blocks that were interrupted will now continue
- Fixed abstract statues not letting people use the radial menu
- Fixed chisel targets to have no range requirement (so you can use binoculars, cameras, etc.)
- Refactor a lot of the chisel and statue code to be more robust
- Add sculpting sounds when chisels are used on carving blocks

## Why It's Good For The Game
Less bugs and now people can now sculpt easier.

## Changelog
:cl:
soundadd: Chisels now make sculpting sounds when used on carving blocks
qol: Chisels now give balloon alerts when selecting a target, cancelling, or sculpting
qol: Chisels used on carving blocks that were interrupted will now continue
fix: Fix chisel targets to have no range requirement (so you can use binoculars, cameras, etc.)
fix: Fix standard mineral blocks (uranium, diamond, plasma, etc.) not letting people use the radial menu to craft abstract statues properly.  This does not work with custom mineral blocks. (pizza, glass, etc.)
code: Change chisel and statue code to be more readable
/:cl:

### CODE REVIEW

File: code/modules/art/statues.dm [hunk 1]
```diff
@@ -359,42 +361,52 @@ Moving interrupts
...
+	stop_sculpting()
+
 /obj/item/chisel/proc/show_generic_statues_prompt(mob/living/user)
 	var/list/choices = list()
 	for(var/statue_path in prepared_block.get_possible_statues())
-		var/obj/structure/statue/S = statue_path
-		choices[statue_path] = image(icon=initial(S.icon),icon_state=initial(S.icon_state))
-	var/choice = show_radial_menu(user, prepared_block , choices, require_near = TRUE)
+		var/obj/structure/statue/abstract_statue = statue_path
+		choices[statue_path] = image(icon=initial(abstract_statue.icon), icon_state=initial(abstract_statue.icon_state))
```
Comment (Time-Green) [inline code/modules/art/statues.dm:400]:
```suggestion
		choices[statue_path] = image(icon = initial(abstract_statue.icon), icon_state = initial(abstract_statue.icon_state))
```

**Review by Time-Green — CHANGES_REQUESTED** [review]
Nice work! God I'm sorry this was left for 2 fucking weeks. Very minor style nitpicks, code looks good otherwise so just ping me on discord when you're done and I'll merge it

File: code/modules/art/statues.dm [hunk 2]
```diff
@@ -348,9 +348,11 @@ Moving interrupts
 	var/interrupted = FALSE
 	var/remaining_time = sculpting_time - (prepared_block.completion * sculpting_time)
 
-	var/datum/progressbar/total_progress_bar = new(user, sculpting_time, prepared_block )
+	var/datum/progressbar/total_progress_bar = new(user, sculpting_time, prepared_block)
 	while(remaining_time > 0 && !interrupted)
-		if(do_after(user,sculpting_period, target = prepared_block, progress = FALSE))
+		if(do_after(user, sculpting_period, target = prepared_block, progress = FALSE))
+			if(!(remaining_time % 4)) // gives a small delay so we don't play this sound every second
```
[the remark this answers is not in the document]
Comment (timothymtorres) [reply code/modules/art/statues.dm]: Done with https://github.com/tgstation/tgstation/pull/72638/commits/c860c6f3517c23d3409ea134ae5e42f1c2a2d9bb

File: code/modules/art/statues.dm [hunk 3]
```diff
@@ -348,9 +348,12 @@ Moving interrupts
 	var/interrupted = FALSE
 	var/remaining_time = sculpting_time - (prepared_block.completion * sculpting_time)
 
-	var/datum/progressbar/total_progress_bar = new(user, sculpting_time, prepared_block )
+	var/datum/progressbar/total_progress_bar = new(user, sculpting_time, prepared_block)
 	while(remaining_time > 0 && !interrupted)
-		if(do_after(user,sculpting_period, target = prepared_block, progress = FALSE))
+		if(do_after(user, sculpting_period, target = prepared_block, progress = FALSE))
+			var/time_delay = !(remaining_time % 4)
```
Comment (Time-Green) [inline code/modules/art/statues.dm:354]: This is not what I meant, it's just the magic '4' number

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/modules/art/statues.dm (hunk 3 above)
Comment (timothymtorres) [reply code/modules/art/statues.dm:354]: The time delay is not in `SECONDS`  or `MINUTES` it is every 4th loop through the `while()` loop.  That's what the modulo operator does with the way I'm using it here.

Comment (Time-Green) [reply code/modules/art/statues.dm:354]: Yes I know, I mean put the 4 on the object

File: code/modules/art/statues.dm [hunk 4]
```diff
@@ -1,3 +1,5 @@
+#define SCULPT_SOUND_INCREMENT 4
```
Comment (Time-Green) [inline code/modules/art/statues.dm:1]: Could you autodoc this?

**Review by Time-Green — CHANGES_REQUESTED** [review]

File: code/modules/art/statues.dm (hunk 3 above)
Comment (timothymtorres) [reply code/modules/art/statues.dm]: https://github.com/tgstation/tgstation/pull/72638/commits/4447d46078003bb9109f6d02b64600b51635a1b3

File: code/modules/art/statues.dm (hunk 4 above)
Comment (timothymtorres) [reply code/modules/art/statues.dm:3]: https://github.com/tgstation/tgstation/pull/72638/commits/25b840b472ca451c556a0a72effe7e7f9a8c9ea5

**Comment by adamkad1** [comment]
When will we be able to paint sculptures in natural colors tho. Why do sculptures turn lame once finished?

**Pull request merged**